Exception occurs when search after index (HNSW)

<!-- Please state your issue using the following template and, most importantly, in English. -->

#### Steps/Code to reproduce:
Run test case:
/milvus/tests20/python_client/testcases/test_search.py::TestCollectionSearch::test_search_after_different_index

Logs:
[log.tar.zip](https://github.com/milvus-io/milvus/files/6860670/log.tar.zip)


#### Expected result:
No exceptions

#### Actual results:
FAILED test_search.py::TestCollectionSearch::test_search_after_different_index[2-8-False-False-HNSW-params4]

(api_response) : <BaseException: (code=1, message=Search failed, C runtime error detected, error code = 1, error msg = [json.exception.type_error.302] type must be number, but is null

#### Environment:
- Milvus version(): 9aa56cd
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2):
Name: pymilvus
Version: 2.0.0rc3.dev6
Name: pymilvus-orm
Version: 2.0.0rc3.dev6

- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

#### Configuration file:


#### Additional context:

--- a/internal/core/src/query/SearchOnSealed.cpp
+++ b/internal/core/src/query/SearchOnSealed.cpp
@@ -15,6 +15,8 @@
 
 #include "query/SearchOnSealed.h"
 #include <knowhere/index/vector_index/VecIndex.h>
+#include "knowhere/index/vector_index/ConfAdapter.h"
+#include "knowhere/index/vector_index/ConfAdapterMgr.h"
 #include "knowhere/index/vector_index/helpers/IndexParameter.h"
 #include "knowhere/index/vector_index/adapter/VectorAdapter.h"
 #include <boost_ext/dynamic_bitset_ext.hpp>
@@ -92,6 +94,9 @@ SearchOnSealed(const Schema& schema,
         auto conf = search_info.search_params_;
         conf[milvus::knowhere::meta::TOPK] = search_info.topk_;
         conf[milvus::knowhere::Metric::TYPE] = MetricTypeToName(field_indexing->metric_type_);
+        auto index_type = field_indexing->indexing_->index_type();
+        auto adapter = milvus::knowhere::AdapterMgr::GetInstance().GetAdapter(index_type);
+        Assert(adapter->CheckSearch(conf, index_type, field_indexing->indexing_->index_mode()));
         return field_indexing->indexing_->Query(ds, conf, bitset);
     }();
 

